validation/chOpenApiValidator.go

```diff
@@ -42,40 +43,41 @@ func NewCHValidator() CHValidator {
 // ValidateRequestAgainstOpenApiSpec takes a request and an openAPI spec location (string relative path) and uses the
 // spec to validate the provided request. If any validation errors are found, then they are formatted and returned to the
 // caller. If any errors are encountered while attempting to validate, they are handled and also returned to the caller.
-func (chv CHValidatorImpl) ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec string) ([]byte, error) {
+func (chv CHValidatorImpl) ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec, contextId string) ([]byte, error) {
 
 	// Get the Open API 3 validation schema location.
 	ctx := context.Background()
 	loader := &openapi3.Loader{Context: ctx, IsExternalRefsAllowed: true}
 	abs, err := callFilepathAbs(openApiSpec)
 	if err != nil {
-		log.Error(fmt.Errorf("error occured while retrieving absolute path of validation schema file: %s", err))
+		log.ErrorC(contextId, err, log.Data{"msg": "error occurred while retrieving absolute path of validation schema file"})
 		return nil, err
 	}
-	log.Info(fmt.Sprintf("Retrieved absolute path of validation schema: %s", abs))
+	log.InfoC(contextId, fmt.Sprintf("Retrieved absolute path of validation schema "), log.Data{config.SchemaAbsolutePath : abs})
 
 	// Load the validation schema.
 	doc, err := loader.LoadFromFile(abs)
 	if err != nil {
-		log.Error(fmt.Errorf("unable to open Open API spec: %s", openApiSpec), nil)
+		log.ErrorC(contextId, err, log.Data{config.OpenApiSpec : openApiSpec, "msg": "unable to open Open API spec"})
 		return nil, err
 	} else {
 		if err := doc.Validate(ctx); err != nil {
-			log.Error(fmt.Errorf("error occured while trying to call kin-openAPI validation method: %s", err))
+			log.ErrorC(contextId, err, log.Data{"msg": "error occurred while trying to call kin-openAPI validation method"})
```

Comment:
Since msg is used in so many places it would help to add a MessageKey to keys.go

Reply:
Added message as a key